Fix not showing extra attributes in assembly when it's empty

andreslucena opened the pull request:
#### :tophat: What? Why?

There's a bug in the Assemblies' main data content block (enabled by default in the seeds), where if some of the extra attributes that are shown in the description are empty, the title is still shown. 

This PR fixes it. It also introduces a spec for this CB that was missing.  

#### Testing

1. Sign in as admin
2. Edit an assembly and remove the "Purpose of action" in the edit form
3. Go to the landing page with the "Main data" content block, and see that the "Purpose of action" isn't shown anymore

### :camera: Screenshots

#### Before
![Screenshot of the assembly landing page with the bug](https://github.com/decidim/decidim/assets/717367/4ea968c4-2173-4869-b06f-0bce14c48e63)

#### After
![Screenshot of the assembly landing page with the bug fixed](https://github.com/decidim/decidim/assets/717367/3368268c-4ba6-4285-a2e0-d15afaee81d8)

:hearts: Thank you!

alecslupu commented on decidim-assemblies/app/cells/decidim/assemblies/content_blocks/main_data_cell.rb:
```diff
@@ -33,7 +33,7 @@ def description_text
         def extra_attributes
           EXTRA_ATTRIBUTES.filter_map do |attribute|
-            next if (text = translated_attribute(send(attribute))).blank?
+            next if (text = translated_attribute(send(attribute))) == "<p></p>"
```
```suggestion
            text =  translated_attribute(send(attribute))
            next if text.blank? || text == "<p></p>" 
```

alecslupu commented on decidim-assemblies/spec/cells/decidim/assemblies/content_blocks/main_data_cell_spec.rb:
```diff
@@ -0,0 +1,109 @@
+      let(:purpose_of_action) { "<p></p>" }
+
+      it_behaves_like "without extra attribute"
+    end
```
```suggestion
    end
    
    context "when there is no extra attribute" do
      it_behaves_like "without extra attribute"
    end
```

andreslucena replied:
Nice! This is done in 01398421ea, so we have it for all the extra attributes.

alecslupu reviewed (changes requested):
@andreslucena This needs to be improved. When working with seed data that usually comes as nil or blank, the fields you want to hide, are being displayed... 

to reproduce this, just visit the frontend interface of an unmodified assembly. 

![image](https://github.com/decidim/decidim/assets/105683/1904a923-7b5a-4e01-87f3-82d3f0ae1947)

andreslucena commented:
> @andreslucena This needs to be improved. When working with seed data that usually comes as nil or blank, the fields you want to hide, are being displayed... 

You're right! This was also happening with a normal behavior: filling a new assembly without touching these fields. The empty HTML paragraph (i.e. `<p></p>`) only happened when you entered some text here and then removed it. 

I've changed the behavior and also added the specs for these cases in the last set of commits.

alecslupu commented:
@andreslucena there is a rubocop failing ... Can u handle that ?

andreslucena commented:
> @andreslucena there is a rubocop failing ... Can u handle that ?

Done in 28b0eeacc0

alecslupu reviewed (approved):
LGTM
